doc/community/developer-guide.md

```diff
@@ -205,4 +205,96 @@ pytest src/tests/integration/test_generic_db_operations.py::test_profile_query
\ No newline at end of file
+Please note that if you submit a pull request from a forked repository, the integration testing phase will be skipped because the pre-registered accounts won't be accessible.
+## General SQL Clause for Multiple Database Dialects
+
+### Context
+
+As our codebase is expanding, we have noticed that we need to write SQL queries for different database dialects such as MySQL, PostgreSQL, SQLite, and more. Writing and maintaining separate queries for each database can be time-consuming and error-prone.
+
+To address this issue, we can use `sqlglot` to create a construct that can be compiled across multiple SQL dialects. This clause will allow us to write a single SQL query that can be translated to different database dialects, then use it for calculating the metadata (e.g. metadata used by boxplot)
+
+In this document, we'll explain how to build generic SQL constructs and provide examples of how it can be used in our codebase. We will also include instructions on how to add support for additional database dialects.
+
+### Approach 1 - Provide the general SQL Clause
+
+We can use [SQLGlot](https://sqlglot.com/sqlglot.html) to build the general sql expressions.
+
+Then transpile to the sql which is supported by current connected dialect.
+
+Our `sql.connection.Connection._transpile_query` will automatically detect the dialect and transpile the SQL clause.
+
++++
+
+#### Example
+
+```{code-cell} ipython3
+from sqlglot import select, condition
+from sql import connection
+from sqlalchemy import create_engine
+
+# Prepare connection
+conn = connection.Connection(engine=create_engine(url="sqlite://"))
+
+where = condition("x=1").and_("y=1")
+general_sql = select("*").from_("y").where(where).sql()
+
+print("General SQL Clause: ")
+print(f"{general_sql}\n")
+
+print("Transpiled result: ")
+conn._transpile_query(general_sql)
+```
+
+### Approach 2 - Provide SQL Clause based on specfic database 
+
+Sometimes the SQL Clause might be complex, we can also write the SQL Clause based on one specfic database and transpile it.
+
+For example, the `TO_TIMESTAMP` keyword is only defined in duckdb, but we want to also apply this SQL clause to other database.
+
+We may provide `sqlglot.parse_one({source_sql_clause}, read={source_database_dialect}).sql()` as input sql to `_transpile_query()`
+
++++
+
+#### When current connection is via duckdb
+
+```{code-cell} ipython3
+from sql import connection
+from sqlalchemy import create_engine
+import sqlglot
+
+# Prepare connection
+conn = connection.Connection(engine=create_engine(url="duckdb://"))
+
+# Prepare SQL clause based on duckdb syntax
+input_sql = sqlglot.parse_one("SELECT TO_TIMESTAMP(1618088028295)", read="duckdb").sql()
+
+print(f"{conn._get_curr_sqlglot_dialect()} transpiled result:")
+conn._transpile_query(input_sql)
```

Comment:
![image](https://user-images.githubusercontent.com/103026689/232441443-e28b8b53-f684-4bd7-9a82-8a5b72dc0e91.png)

same as before. I think we can remove the print statement OR include `conn._transpile_query(input_sql)` in the print OR out the last command `conn._transpile_query(input_sql)` in a new cell with title `duckdb transpiled result`